### ISSUE TITLE
Android cmdline Builds + Android Code Doesn't Build

### ISSUE THREAD

**Issue by ashes999** [Kode/Kha#533]
TL;DR: 

- I would like to see one-click (or one-command) android builds that generate APKs. Right now, it's two steps.
- I currently get [this exact error](http://stackoverflow.com/questions/40927360/java-lang-unsupportedclassversionerror-com-android-build-gradle-appplugin-uns/42681017#42681017) when I build the Android version, which I can resolve by changing the Gradle dependency version to `2.2.2` (see that answer in my question)

Longer version: building an APK for my Kha app is two steps:

- `kha android`
- `cd build/android`; `./gradlew assemble`

The second step fails (see above TL;DR comment). Is this something I did wrong? It looks like the generated Gradle app uses an incorrect/incompatible version of the `com.android.tools.build` artefact.

Second: is it possible to update Kha's build system to either always build the APK, or to expose a command-line parameter for this?

If you have an Android-specific issue, your development cycle is pretty painful:

- Make a change
- Build via `kha android`
- Open in the Android IDE
- Build in the IDE (or `./gradlew`)
- Deploy the APK to your device and test

If you have to do this in quick succession, it gets tedious really fast.

**Comment by RobDangerous** [comment]
This is the ideal workflow IMHO because it enables debugging support in Android Studio. You don't need to close and reopen Android Studio when you do a change by the way. Nonetheless we can of course add the gradle call behind the --compile parameter like we do for other targets.
Your error is probably due to the version of Android Studio being newer than whatever I used when I worked on it the last time - I don't work on it too often because there's also the android-native target which most people use.

**Comment by ashes999** [comment]
I didn't know about the `android-native` target. I think I have the wrong mindset (things regenerate every time) which doesn't match how Kha works. Debugging, I agree, is awesome (but I usually use `trace` statements to debug).

I'll close this based on your first sentence `This is the ideal workflow [...]`. Thanks for the feedback though.

**Comment by sh-dave** [comment]
+1 for handling the --compile option like almost every other target does.

**Comment by ashes999** [comment]
Forgot to re-open and discuss: what about the Gradle build dependency change?

**Comment by RobDangerous** [comment]
Gradle settings are now up to date.

**Comment by RobDangerous** [comment]
--compile support is here for android and android-native.
